apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +124,54 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp calculate_package_total(
+         %{tax_total: tax_total, adjustment_total: adjustment_total, promo_total: promo_total},
+         shipping_cost
+       ) do
+    Enum.reduce([tax_total, adjustment_total, promo_total, shipping_cost], &Money.add!/2)
+  end
+
+  def process_package(package) do
+    shipping_method =
+      Enum.find(package.shipping_methods, fn %{id: id} ->
+        id == package.shipping_method_id
+      end)
+    shipping_cost = shipping_method.cost
+
+    package_total =
+      Enum.reduce(
+        [package.tax_total, package.adjustment_total, package.promo_total, shipping_cost],
+        &Money.add!/2
+      )
+
+    Package.update(package, %{cost: shipping_cost, total: package_total, shipping_method_id: shipping_method.id})
+  end
+
+  @doc """
+  Persists shipping_method_id to packages
+
+  Calculate pacakge total cost Sum(shipping_cost, adjustment_total,promo_total, shipping_cost)
+
+  Update package cost total in DB
+  """
+
+  @spec associate_package(Context.t()) :: Context.t()
+  def associate_package(%Context{valid?: true, struct: %Order{} = order} = context) do
```

Comment:
Where are the `form_params` that we discussed about (so clearly) in the [meeting](https://hackmd.io/atLBxkpvReiEwSbx5CzUTA#Live-coding)? Shouldn't they be a part of the context?
Why is this function named `associate_packages`? The packages are already associated with the order :thinking: